presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -314,6 +316,32 @@ public HiveClientConfig setMetastoreRefreshInterval(Duration metastoreRefreshInt
         return this;
     }
 
+    public long getMetastoreCacheMaximumSize()
+    {
+        return metastoreCacheMaximumSize;
+    }
+
+    @Min(1)
+    @Config("hive.metastore-cache-maximum-size")
+    public HiveClientConfig setMetastoreCacheMaximumSize(long metastoreCacheMaximumSize)
+    {
+        this.metastoreCacheMaximumSize = metastoreCacheMaximumSize;
+        return this;
+    }
+
+    public long getMetastorePerTransactionCacheMaximumSize()
+    {
+        return metastorePerTransactionCacheMaximumSize;
+    }
+
+    @Min(1)
+    @Config("hive.metastore-per-transaction-cache-maximum-size")
```

Comment:
I think this might read better as `hive.per-transaction-metastore-cache-maximum-size`. That way if we add other per-transaction properties they will all have the same prefix. Up to you though.